Test issues at tearDown

I notices this issues on Flint tests:

It is maybe relevant for silx
```
___________________ TestLiveWindow.test_feed_default_layout ____________________
/opt/conda/envs/testenv/lib/python3.7/site-packages/silx/gui/utils/testutils.py:199: in tearDown
    self._checkForUnreleasedWidgets()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <test_live_window.TestLiveWindow testMethod=test_feed_default_layout>
    def _checkForUnreleasedWidgets(self):
        """Test fixture checking that no more widgets exists."""
        gc.collect()
    
        widgets = [widget for widget in self.qapp.allWidgets()
                   if (widget not in self.__previousWidgets and
                       _inspect.createdByPython(widget))]
        del self.__previousWidgets
    
        if qt.BINDING in ('PySide', 'PySide2'):
            return  # Do not test for leaking widgets with PySide
    
        allowedLeakingWidgets = self.allowedLeakingWidgets
        self.allowedLeakingWidgets = 0
    
        if widgets and len(widgets) <= allowedLeakingWidgets:
            _logger.info(
                '%s: %d remaining widgets after test' % (self.id(),
                                                         len(widgets)))
    
        if len(widgets) > allowedLeakingWidgets:
            raise RuntimeError(
>               "Test ended with widgets alive: %s" % str(widgets))
E           RuntimeError: Test ended with widgets alive: [<silx.gui.plot.backends.BackendMatplotlib.BackendMatplotlibQt object at 0x7f5f81a3d7d0>]
/opt/conda/envs/testenv/lib/python3.7/site-packages/silx/gui/utils/testutils.py:191: RuntimeError
```

```
______________________ TestLiveWindow.test_layout_actions ______________________
self = <test_live_window.TestLiveWindow testMethod=test_layout_actions>
    def tearDown(self):
        if len(self.__class__._exceptions) > 0:
            messages = "\n".join(self.__class__._exceptions)
>           raise AssertionError("Exception occured in Qt thread:\n" + messages)
E           AssertionError: Exception occured in Qt thread:
E           Traceback (most recent call last):
E             File "/opt/conda/envs/testenv/lib/python3.7/site-packages/silx/gui/plot/backends/BackendMatplotlib.py", line 1333, in leaveEvent
E               self._plot.onMouseLeaveWidget()
E             File "/opt/conda/envs/testenv/lib/python3.7/site-packages/silx/gui/plot/backends/BackendBase.py", line 76, in _plot
E               raise RuntimeError('This backend is no more attached to a Plot')
E           RuntimeError: This backend is no more attached to a Plot
/opt/conda/envs/testenv/lib/python3.7/site-packages/silx/gui/utils/testutils.py:196: AssertionError
```

--- a/silx/gui/plot/backends/BackendMatplotlib.py
+++ b/silx/gui/plot/backends/BackendMatplotlib.py
@@ -1330,7 +1330,12 @@ class BackendMatplotlibQt(FigureCanvasQTAgg, BackendMatplotlib):
 
     def leaveEvent(self, event):
         """QWidget event handler"""
-        self._plot.onMouseLeaveWidget()
+        try:
+            plot = self._plot
+        except RuntimeError:
+            pass
+        else:
+            plot.onMouseLeaveWidget()
 
     # picking
 

